src/main/java/com/cleanroommc/groovyscript/mapper/BlockStateMapper.java

```diff
@@ -0,0 +1,49 @@
+package com.cleanroommc.groovyscript.mapper;
+
+import com.cleanroommc.groovyscript.api.Result;
+import com.cleanroommc.groovyscript.compat.mods.GroovyContainer;
+import com.cleanroommc.groovyscript.server.Completions;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.init.Blocks;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.fml.common.registry.ForgeRegistries;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collections;
+import java.util.List;
+
+public class BlockStateMapper extends AbstractObjectMapper<IBlockState> {
+
+    public static final BlockStateMapper INSTANCE = new BlockStateMapper("blockstate", null);
```

Comment:
since this is the only place you are going to init `BlockStateMapper`, why pass `null` here and not drop and argument and call `super(name, null, IBlockState.class)`?

Reply:
I was doing that for item so that mods can extend that class for a template (for example gregtech metaitems). I guess its not very useful for blockstate